src/languageclient.ts

```diff
@@ -0,0 +1,153 @@
+                configurationSection: 'terraformLanguageServer',
+                // Notify the server about file changes to '.clientrc files contained in the workspace
+                fileEvents: workspace.createFileSystemWatcher('**/*.tf'),
+            }
+        };
+
+        // Create the language client and start the client.
+        let l = new LanguageClient(
+            'terraformLanguageServer',
+            'Language Server',
+            serverOptions,
+            clientOptions,
+            true
+        );
+
+        l.trace = 2;
+
+        l.onReady().then(() => {
+            const capabilities = l.initializeResult && l.initializeResult.capabilities;
+            if (!capabilities) {
+                return vscode.window.showErrorMessage('The language server is not able to serve any features at the moment.');
+            }
+        });
+
+        // Start the client. This will also launch the server
+        ctx.subscriptions.push(l.start());
+    }
+
+    public async installLanguageServer(platform: string, path: string): Promise<void> {
+        let continueInstall = await vscode.window.showInformationMessage(
+            "Would you like to install Terraform Language Server from juliosueiras/terraform-lsp? \n This provides experimental Terraform 0.12 support",
+            "Install", "Abort");
+        if (continueInstall === "Abort") {
+            vscode.window.showErrorMessage("Terraform language server install aborted");
+            return;
+        }
+        const downloadedZipFile = Path.join(path, 'terraform-lsp.tar.gz');
+        // Download language server
+        const octokit = new Octokit();
+
+        // What releases are available?
+        const availableReleses = await octokit.repos.listReleases({
+            owner: 'juliosueiras',
+            repo: 'terraform-lsp'
+        })
+        let releaseOptions = [];
+        availableReleses.data.forEach(r => {
+            releaseOptions.push({
+                label: r.name,
+                description: `Is Prerelease version?: ${r.prerelease} Tag: ${r.tag_name}`,
+                detail: r.id.toString()
+            } as QuickPickItem);
+        });
+
+        let choice = await vscode.window.showQuickPick(releaseOptions, { placeHolder: "Terraform language server install - please pick a version" })
+        if (choice === null) {
+            vscode.window.showErrorMessage("You must pick a version to complete installation");
+            return;
+        }
+
+        const release = await octokit.repos.getRelease({
+            owner: 'juliosueiras',
+            repo: 'terraform-lsp',
+            release_id: choice.detail
+        })
+
+        let downloadUrl = "";
+        release.data.assets.forEach(asset => {
+            if (!asset.name.endsWith('.tar.gz')) {
+                return;
+            }
+
+            if (asset.name.includes(platform)) {
+                downloadUrl = asset.browser_download_url;
+            }
```

Comment:
This currently doesn't work for Windows - the node process platform is `win32`, but the LSP release artifact only has `windows` in the name.

Reply:
Ah man feel silly, should have spotted that one. I've added a fix and a runtime check to report and error if this occurs for a platform or we introduce a regression.

Reply:
Hopefully not all fixed :)